### PR TITLE
[docs] Update custom field pickers using theme scoping

### PR DESCRIPTION
**Pull request by siriwatknp**
With [theme scoping](https://github.com/mui/material-ui/pull/36664), using Material UI and Joy UI together is simple. The version needs `5.12.0`.

DataGrid integration with Joy UI would benefit from this PR as well.

### CODE REVIEW

**Comment by flaviendelangle** [comment]
@siriwatknp I fixed the CI, except for the regressions test which seems to be an actual issue when running `PickerWithJoyField` inside the regression tests :/

File: packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts [hunk 1]
```diff
@@ -271,18 +271,25 @@ export const usePickerValue = <
...
+      lastControlledValue: inValue,
+      ...(isUpdateComingFromPicker
+        ? {}
+        : {
+            lastCommittedValue: inValue,
+            lastPublishedValue: inValue,
+            draft: inValue,
+            hasBeenModifiedSinceMount: true,
+          }),
+    }));
```
Comment (siriwatknp) [inline packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts:292]: @flaviendelangle Is this required for this PR?

Comment (LukasTy) [reply packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts:292]: Oh no, there is a separate PR for it. 🙈 
https://github.com/mui/mui-x/pull/8674

Comment (flaviendelangle) [reply packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts:292]: Looks like I failed my git merge somehow...
I'll re-merge on master once #8674 is merged

Comment (flaviendelangle) [reply packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts]: Done :+1:

**Review by flaviendelangle — APPROVED** [review]

**Pull request merged**